addon/components/model-form.js

```diff
@@ -65,9 +66,16 @@ export default class ModelFormComponent extends Component {
     // set here...
     const deepMerge = (original, objectToApply) => {
       Object.entries(objectToApply).forEach(([key, value]) => {
-        if (value !== null && value !== undefined) {
+        if (
+          original !== null &&
+          original !== undefined &&
+          value !== null &&
+          value !== undefined
+        ) {
           typeof value === "object"
             ? deepMerge(original[key], objectToApply[key])
+            : this.gwr.isIsoDate(value) // Date casts necessary for import
```

Comment:
As discussed, maybe put this in a `utils.js` file?

Reply:
done